lib/types/scan.ts

```diff
@@ -0,0 +1,56 @@
+// Core scan types for index scanning functionality
+export type ScanType = 'single' | 'range'
+export type RangeField = 'start' | 'end'
+
+// Index field types for better type safety
+export type IndexField = 'account' | 'address'
+
+// Enum alternatives (provide runtime values if needed)
+export enum ScanTypeEnum {
+  SINGLE = 'single',
+  RANGE = 'range',
+}
+
+export enum RangeFieldEnum {
+  START = 'start',
+  END = 'end',
+}
```

Comment:
Being this an enum, it could be used when asking for equallity, or setting it as value, instead of just using the string value around the code